crypto/src/main/scala/org/bitcoins/crypto/XOnlyPubKey.scala

```diff
@@ -22,7 +22,49 @@ case class XOnlyPubKey(bytes: ByteVector) extends NetworkElement {
     SchnorrPublicKey(bytes)
   }
 
+  def publicKey: ECPublicKey = schnorrPublicKey.publicKey
+
   def coord: CurveCoordinate = CurveCoordinate(bytes)
+
+  /** @see https://github.com/bitcoin/bitcoin/blob/3340d46cd363e568ce842b2a9930e30902d150ca/src/pubkey.cpp#L227
+    * @see https://github.com/bitcoin-core/secp256k1/blob/9a5a87e0f1276e0284446af1172056ea4693737f/src/modules/extrakeys/main_impl.h#L151
+    * @param pubKey the internal tapscript pubkey
+    * @param merkleRootOpt the merkle root of the tapscript tree, if empty means we have no scripts in the tapscript tree
+    * @param parity the expected parity of the public key reproduced
+    * @return
+    */
+  def checkTapTweak(
```

Comment:
these functions feel like they belong somewhere, not on `XOnlyPubKey`

Reply:
This is where it is in bitcoin core: 

https://github.com/bitcoin/bitcoin/blob/df933596e7e9aa17f7e5cd6e1c850520f5b56f1b/src/pubkey.cpp#L227